.github/actions/test/action.yml

```
@@ -0,0 +1,35 @@
+name: "Test on Linux"
+description: "Run the unit tests on Linux"
+inputs:
+  build-artifact:
+    description: "Name of the build artifact that contains the unit test"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Download build test directory
+      uses: actions/download-artifact@v3
+      with:
+        name: ${{ inputs.build-artifact }}
+    - name: extract build archive
+      shell: bash
+      run: |
+        tar -xzf build.tar.gz
+    - name: run unit test
+      shell: bash
+      run: |
+        cd build/tests
+        sudo chmod a+x gmgpolar_tests
+        ./gmgpolar_tests --gtest_output="xml:testreport.xml"
+    - name: Upload test report
+      uses: actions/upload-artifact@v3
+      with:
+        name: test-report
+        path: build/tests/testreport.xml
+        if-no-files-found: error
+        retention-days:
+    - name: clean-up
+      uses: geekyeggo/delete-artifact@v1
+      with:
+        name: ${{ inputs.build-artifact }} 
```

```suggestion
    - name: clean-up
      uses: geekyeggo/delete-artifact@v1
      with:
        name: ${{ inputs.build-artifact }} 
```

I think if we do not clean up, we can retain it and download it for some days here for example: https://github.com/mknaranja/GMGPolar/actions/runs/4434953337?pr=12